pypeit/pypeit.py

```diff
@@ -482,16 +529,71 @@ def init_one_science(self, sci_ID, det):
         sci_image_files = self.fitstbl.find_frame_files('science', sci_ID=sci_ID)
         scidx = self.fitstbl.find_frames('science', sci_ID=sci_ID, index=True)[0]
         self.sciI = scienceimage.ScienceImage(self.spectrograph, sci_image_files, det=det,
+                                              binning = self.fitstbl['binning'][scidx],
                                               objtype='science', scidx=scidx, setup=self.setup,
                                               par=self.par['scienceimage'],
                                               frame_par=self.par['scienceframe'])
         msgs.sciexp = self.sciI  # For QA on crash
 
         # Names and time
-        self.obstime, self.basename = self.sciI.init_time_names(self.fitstbl)
+        self.obstime, self.basename = self.init_time_names(self.fitstbl, scidx)
         # Return
         return self.obstime, self.basename  # For fluxing
 
+    # Move out of the scienceimage class to here. It is more appropriate here where fitstable exists
```

Comment:
Hmm, but doesn't it refer to a specific science frame?
If so, it is more appropriate in the scienceimage class.
`fitstbl` is irrelevant.